BitSea.cpp

```diff
@@ -0,0 +1,293 @@
+#include "BitSea.hpp"
+
+int main(int argc, char *argv[]) {
+	BitSea bitsea(argc, argv);
+	bitsea.run();
+	
+	return 0;
+}
+
+BitSea::BitSea(int argc, char **argv) : io_service(new boost::asio::io_service), work(new boost::asio::io_service::work(*io_service)) {		
+	tStats.downloaded = 0;
+	tStats.uploaded = 0;
+	tStats.left = 0;
+	
+	cli::parseCommandLine(argc, argv, programSettings);
+	torrentInfo = new TorrentFileParser(programSettings.fileName);
+	tStats.pieceLength = torrentInfo->info.getPieceLength();
+	
+	allocatedPieces = new bool[tStats.numberOfPieces];
+}
+
+void BitSea::run() {
+	fileList = allocateStorage();
+	initPieceDatabase();
+	boost::shared_ptr<Tracker> trackMan = initTracker();
+	
+	// Create worker threads for boost asio to farm jobs out to.
+	for(int i=0; i < THREAD_MAX; i++)
+		worker_threads.create_thread(boost::bind(&BitSea::WorkerThread, this, io_service));
+	
+	startTrackerUpdater(io_service, trackMan);
+	peerList = trackMan->getPeers();
+	
+	// Connect to each peer given by the tracker.
+	for(int i=0; i < peerList.size(); i++) {
+		std::pair<std::string, int> peerPair(peerList[i].peer_id, i);
+		peerResolver.insert(peerPair);
+		talkToPeer(peerList[i], trackMan->getPeerId(), i);
+	}
+    worker_threads.join_all();
```

Comment:
To improve style/readability, I recommend running clang-format:
http://clang.llvm.org/docs/ClangFormat.html

And cpplint:
https://en.wikipedia.org/wiki/Cpplint